website/client/src/components/chat/autoComplete.vue

```diff
@@ -90,12 +90,14 @@ export default {
   props: ['selections', 'text', 'caretPosition', 'coords', 'chat', 'textbox'],
   data () {
     return {
-      atRegex: /(?!\b)@[\w-]*$/,
+      // Old Regex
+      // atRegex: /(?!\b)@([\w-]*)$/,
+      atRegex: /@([\w-]*)$/,
```

Comment:
Why remove the negative word boundary lookahead?  I agree that a positive lookbehind for negated word boundary would probably make more sense but the search should not start when typing e.g. an e-mail address right?

Ah is this to enable searches anywhere in the text?  If that's the case you should drop the end-of-text anchor, however that worries me that you might find earlier mentions in the same text block with this regex and the selection will jump back to that mention instead.  I didn't run your branch yet, so not sure, but would be good to check.

Also it's not necessary to leave old code around.  It's all in the git history anyway.

Reply:
Hehe... my secret weapon... XD

First of all, negative lookahead is awkward since the mentions trigger without boundaries, so typing an email is a byproduct. If the user is annoyed, they can escape the search (or the search will pop up invalid).

The end of line is necessary to find the last mention of the textblock.

Reply:
I'm not sure what you mean, the whole point of the negative lookahead is to make the mentions trigger only without boundaries right?  Historically the choice was made to favour not launching searches in e-mails over being able to type two mentions after each other without spaces.

I'm not sure why that behaviour should change now.  Was that discussed in the issue perhaps?  I do agree that the negative lookahead is not necessary, since the word boundary match is a zero-character match anyway. `/\B@[\w-]*$/` should have the exactly the same behaviour.

Or am I missing something?

Reply:
Oh dang! Where is the history?

And no it's slightly different behavior... the search will include or not include the word boundary.